src/MLOps.NET/Storage/Repositories/DataRepository.cs

```diff
@@ -71,5 +73,67 @@ public Data GetData(Guid runId)
 
             return data;
         }
+
+        ///<inheritdoc cref="IDataRepository"/>
+        public async Task LogDataDistribution<T>(Guid runId, IDataView dataView, string columnName) where T : struct
+        {
+            using var db = this.contextFactory.CreateDbContext();
+
+            var data = db.Data
+                .Include(d => d.DataSchema.DataColumns)
+                .First(d => d.RunId == runId);
+
+            var dataColumn = data.DataSchema
+                .DataColumns
+                .FirstOrDefault(c => c.Name == columnName);
+
+            var column = dataView.Schema.First(c => c.Name == columnName);
+
+            List<DataDistribution> list = GetDataDistributionForColumn<T>(dataView, column, dataColumn);
+
+            db.DataDistributions.AddRange(list);
+
+            await db.SaveChangesAsync();
+        }
+
+        ///<inheritdoc cref="IDataRepository"/>
+        public List<DataDistribution> GetDataDistribution<T>(Guid runId, IDataView dataView, string columnName) where T:struct
+        {
+            using var db = this.contextFactory.CreateDbContext();
+
+            var data = db.Data
+                .Include(d => d.DataSchema.DataColumns)
+                .First(d => d.RunId == runId);
+
+            var dataColumn = data.DataSchema
+                .DataColumns
+                .FirstOrDefault(c => c.Name == columnName);
+
+            var column = dataView.Schema.First(c => c.Name == columnName);
+
+            return GetDataDistributionForColumn<T>(dataView, column, dataColumn);
+        }
+
+        private List<DataDistribution> GetDataDistributionForColumn<T>(IDataView dataView, DataViewSchema.Column column, Entities.Impl.DataColumn dataColumn) where T : struct
```

Comment:
Could we move this to a separate service, e.g. `DataDistributionCalculator` which is passed into this repository. Like that you can add a lot more unit tests around this method to ensure that it does correctly calculate the distribution for various input data